File: GanttChartLightLibraryDemos/Demos/Samples/WPF-CSharp/ScheduleChartDataGrid/MainFeatures/CustomGanttChartItem.cs
using DlhSoft.Windows.Controls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Windows.Media;

namespace Demos.WPF.CSharp.ScheduleChartDataGrid.MainFeatures
{
    class CustomGanttChartItem : GanttChartItem, INotifyPropertyChanged
    {
        private ImageSource icon;
        public ImageSource Icon
        {
            get { return icon; }
            set
            {
                icon = value;
                OnPropertyChanged("Icon");
            }
        }

        private string note;
        public string Note
        {
            get { return note; }
            set
            {
                note = value;
                OnPropertyChanged("Note");
            }
        }

        protected override void OnPropertyChanged(string propertyName)
        {
            base.OnPropertyChanged(propertyName);
        }
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples/WPF-VisualBasic/ScheduleChartDataGrid/MainFeatures/CustomGanttChartItem.vb
Imports System.ComponentModel
Imports System.Windows.Media
Imports DlhSoft.Windows.Controls

Friend Class CustomGanttChartItem
    Inherits GanttChartItem
    Implements INotifyPropertyChanged

    Private _icon As ImageSource
    Public Property Icon As ImageSource
        Get
            Return _icon
        End Get
        Set(value As ImageSource)
            _icon = value
            OnPropertyChanged("Icon")
        End Set
    End Property

    Private _note As String
    Public Property Note As String
        Get
            Return _note
        End Get
        Set(value As String)
            _note = value
            OnPropertyChanged("Note")
        End Set
    End Property

    Protected Overrides Sub OnPropertyChanged(propertyName As String)
        MyBase.OnPropertyChanged(propertyName)
    End Sub
End Class
